packages/api/src/hooks/p2p/useP2PAdvertCreate.ts

```diff
@@ -0,0 +1,38 @@
+import { useCallback } from 'react';
+import useMutation from '../../useMutation';
+import useInvalidateQuery from '../../useInvalidateQuery';
+
+type TPayload = Parameters<ReturnType<typeof useMutation<'p2p_advert_create'>>['mutate']>[0]['payload'];
+
+/** A custom hook that creates a P2P advert. This can only be used by an approved P2P advertiser.
+ * 
+ * To create an advert, specify the following payload arguments in the `mutate` call (some arguments are optional):
+ * @example
+ *  mutate({
+        description: 'Please transfer to account number 1234',
+        type: 'buy',
+        amount: 100,
+        max_order_amount: 50,
+        min_order_amount: 20,
+        payment_method: 'bank_transfer',
+        rate: 4.25,
+    });
+ * 
+*/
+const useP2PAdvertCreate = () => {
+    const invalidate = useInvalidateQuery();
+    const { mutate: _mutate, ...rest } = useMutation('p2p_advert_create', {
+        onSuccess: () => {
+            invalidate('p2p_advert_list');
+        },
+    });
+
+    const mutate = useCallback((payload: TPayload) => _mutate({ payload }), [_mutate]);
+
+    return {
```

Comment:
@adrienne-deriv based on our discussion do we need to return the response data as well?